cmd/chartsvc/handler.go

```diff
@@ -200,6 +204,90 @@ func getChartVersionValues(w http.ResponseWriter, req *http.Request, params Para
 	w.Write([]byte(files.Values))
 }
 
+func isChartContained(chartData *models.Chart, chartMetadata *chart.Metadata) bool {
+	eq := true
+	// Compare simple fields
+	if chartData.Description != chartMetadata.Description ||
+		chartData.Home != chartMetadata.Home ||
+		chartData.Icon != chartMetadata.Icon {
+		eq = false
+	}
+	// Compare array of key words (but it may be empty)
+	if len(chartData.Keywords) > 0 && len(chartMetadata.Keywords) > 0 {
+		if !reflect.DeepEqual(chartData.Keywords, chartMetadata.Keywords) {
+			eq = false
+		}
+	}
+	// Compare array of maintainers (but it may be empty)
+	if len(chartData.Maintainers) > 0 && len(chartMetadata.Maintainers) > 0 {
+		// Maintainers format is different (pointers versus structs)
+		c2MaintainerArray := []chart.Maintainer{}
+		for _, m := range chartMetadata.Maintainers {
+			c2MaintainerArray = append(c2MaintainerArray, *m)
+		}
+		if !reflect.DeepEqual(chartData.Maintainers, c2MaintainerArray) {
+			eq = false
+		}
+	}
+	if eq {
+		// Charts matches, look if the version exists
+		for _, v := range chartData.ChartVersions {
+			if v.Version == chartMetadata.Version && v.AppVersion == chartMetadata.AppVersion {
+				return true
+			}
+		}
+	}
+	return false
+}
+
+// resolveRepo returns the repo that contains the given chart and the latest version found
+func resolveRepo(w http.ResponseWriter, req *http.Request) {
+	db, closer := dbSession.DB()
+	defer closer()
+
+	metadataBytes, err := ioutil.ReadAll(req.Body)
+	if err != nil {
+		errStr := fmt.Sprintf("could not read body %v", err)
+		log.WithError(err).Errorf(errStr)
+		response.NewErrorResponse(http.StatusUnprocessableEntity, errStr).Write(w)
+		return
+	}
+	chartMetadata := &chart.Metadata{}
+	err = json.Unmarshal(metadataBytes, chartMetadata)
+	if err != nil {
+		errStr := fmt.Sprintf("could not read body %v", err)
```

Comment:
This error message might not be specific enough since it is the same one than before.